src/twisted/internet/defer.py

```diff
@@ -939,38 +951,27 @@ def __str__(self) -> str:
 
     __repr__ = __str__
 
-    def __iter__(self) -> "Deferred[_DeferredResultT]":
-        return self
-
     @_extraneous
-    def send(self, value: object = None) -> "Deferred[_DeferredResultT]":
-        if self.paused:
-            # If we're paused, we have no result to give
-            return self
-
-        result = getattr(self, "result", _NO_RESULT)
-        if result is _NO_RESULT:
-            return self
-        if isinstance(result, Failure):
-            # Clear the failure on debugInfo so it doesn't raise "unhandled
-            # exception"
-            assert self._debugInfo is not None
-            self._debugInfo.failResult = None
-            result.value.__failure__ = result
-            raise result.value
-        else:
-            raise StopIteration(result)
-
-    # For PEP-492 support (async/await)
-    # type note: base class "Awaitable" defined the type as:
-    #     Callable[[], Generator[Any, None, _DeferredResultT]]
-    #     See: https://github.com/python/typeshed/issues/5125
-    #     When the typeshed patch is included in a mypy release,
-    #     this method can be replaced by `__await__ = __iter__`.
-    def __await__(self) -> Generator[Any, None, _DeferredResultT]:
-        return self.__iter__()  # type: ignore[return-value]
+    def __iter__(
+        self,
+    ) -> "Generator[Deferred[_DeferredResultT], _DeferredResultT, _DeferredResultT]":
+        # Wrap failures in an _Outcome object here to avoid buggy .throw()
+        #   https://bugs.python.org/issue29587
+        #   https://bugs.python.org/issue29590
+        # https://github.com/python-trio/trio/blob/ab75cd12a4db68e2fbb2e58f46900ebaa582aab2/trio/_core/_run.py#L2160-L2166
+        return (yield self.addBoth(_Outcome)).unwrap()  # type: ignore[no-any-return,attr-defined,arg-type]
```

Comment:
It seems like this should be a `cast` instead of all these `type:ignore`s? Any `type: ignore` should be scoped as tightly as possible to a single expression